### PR TITLE
Feature/adjust application requests

### PR DESCRIPTION
**Pull request by thebino**
## 📑 What does this PR do?

adjusted all application related requests with tests

# ✅ Checklist

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiImpl.kt [hunk 1]
```diff
@@ -58,4 +44,20 @@ class MastodonApiImpl(
                 }.body()
         }
     }
+
+    override suspend fun verifyApplication(): Result<Application> {
+        return runCatchingIgnoreCancelled<Application> {
+            httpClient.get("/api/v1/apps/verify_credentials") {
+                headers {
+                    append(HttpHeaders.Authorization, "testAuthorization")
```
Comment (crocsandcoffee) [inline data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiImpl.kt:52]: is this for debugging or the actual string that needs to be passed up?

File: data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiImpl.kt [hunk 2]
```diff
@@ -58,4 +44,20 @@ class MastodonApiImpl(
...
+                }
+            }.body()
+        }
+    }
+
+    override suspend fun getInstance(domain: String?): Result<Instance> {
```
Comment (crocsandcoffee) [inline data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiImpl.kt:58]: `domain` shouldn't be null here I believe. but we can fix this later when we hook this up

File: data/network/src/commonMain/kotlin/social/androiddev/common/network/model/Application.kt [hunk 3]
```diff
@@ -17,15 +17,24 @@ import kotlinx.serialization.Serializable
...
     val name: String,
+    @SerialName("vapid_key") val vapidKey: String,
+
+    // optional attributes
+    val website: String? = null,
+)
+
+@Serializable
+data class NewOauthApplication(
+    val id: String,
```
Comment (crocsandcoffee) [inline data/network/src/commonMain/kotlin/social/androiddev/common/network/model/Application.kt:30]: probably good idea to break this out into it's own file with a javadoc on both of these models

**Review by crocsandcoffee — APPROVED** [review]

File: data/network/src/commonMain/kotlin/social/androiddev/common/network/model/Application.kt (hunk 3 above)
Comment (thebino) [reply data/network/src/commonMain/kotlin/social/androiddev/common/network/model/Application.kt:30]: javadoc yes, but a separate file will only bloat it.
In the API Doc the application is re-used. So you'll probably first have a look into this file

**Pull request merged**